FEXCore/Source/Interface/IR/Passes/x87StackOptimizationPass.cpp

```diff
@@ -0,0 +1,1323 @@
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+      // Unary Operations
+      case OP_F80SQRTSTACK: {
+        HandleUnop(OP_VFSQRT, true, OP_F80SQRT);
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+      case OP_F80SINSTACK: {
+        HandleUnop(OP_F64SIN, false, OP_F80SIN);
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+      case OP_F80COSSTACK: {
+        HandleUnop(OP_F64COS, false, OP_F80COS);
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+      case OP_F80F2XM1STACK: {
+        HandleUnop(OP_F64F2XM1, false, OP_F80F2XM1);
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+
+      case OP_F80PTANSTACK: {
+        HandleUnop(OP_F64TAN, false, OP_F80TAN);
+        Ref OneConst {};
+        if (ReducedPrecisionMode) {
+          OneConst = IREmit->_VCastFromGPR(8, 8, GetConstant(0x3FF0000000000000));
+        } else {
+          OneConst = IREmit->_LoadNamedVectorConstant(16, NamedVectorConstant::NAMED_VECTOR_X87_ONE);
+        }
+
+        if (SlowPath) {
+          UpdateTopForPush_Slow();
+          StoreStackValueAtTop_Slow(OneConst);
+        } else {
+          StackData.push(StackMemberInfo {.SourceDataSize = ReducedPrecisionMode ? OpSize::i64Bit : OpSize::i128Bit,
+                                          .SourceDataNode = nullptr,
+                                          .StackDataNode = OneConst,
+                                          .InterpretAsFloat = false});
+        }
+
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+        // Miscellaneous Ops
+        // These are handled on a case-by-case basis
+      case OP_F80SINCOSSTACK: {
+        const auto& [Valid, StackMember] = StackData.top();
+
+        MigrateToSlowPathIf(Valid != StackSlot::VALID);
+
+        // Grab st0
+        Ref st0 {};
+        if (SlowPath) {
+          st0 = LoadStackValueAtTop_Slow();
+        } else {
+          st0 = StackMember.StackDataNode;
+        }
+
+        // Calc values
+        Ref SinValue {};
+        Ref CosValue {};
+        if (ReducedPrecisionMode) {
+          SinValue = IREmit->_F64SIN(st0);
+          CosValue = IREmit->_F64COS(st0);
+        } else {
+          SinValue = IREmit->_F80SIN(st0);
+          CosValue = IREmit->_F80COS(st0);
+        }
+
+        // Push values
+        if (SlowPath) {
+          StoreStackValueAtTop_Slow(SinValue, false);
+          UpdateTopForPush_Slow();
+          StoreStackValueAtTop_Slow(CosValue, true);
+        } else {
+          StackData.setTop(StackMemberInfo {
+            .SourceDataSize = StackMember.SourceDataSize, .SourceDataNode = nullptr, .StackDataNode = SinValue, .InterpretAsFloat = true});
+          StackData.push(StackMemberInfo {
+            .SourceDataSize = StackMember.SourceDataSize, .SourceDataNode = nullptr, .StackDataNode = CosValue, .InterpretAsFloat = true});
+        }
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+      case OP_INITSTACK: {
+
```

Comment:
drop this blank line (there are a handful of cases like this throughout the patch)

Reply:
Think this is sorted.